File: app/src/main/java/de/tum/in/tumcampus/models/ExamList.java
package de.tum.in.tumcampus.models;

import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Root;

import java.util.List;

/**
 * Wrapper class holding a list of exams.
 * <p>
 * Note: This model is based on the TUMOnline web service response format for a
 * corresponding request.
 */
@Root(name = "rowset")
public class ExamList {

	@ElementList(inline = true)
	private List<Exam> grades;

	public List<Exam> getExams() {
		return grades;
	}

	public void setExams(List<Exam> grades) {
		this.grades = grades;
	}

}

File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/grades/model/ExamList.kt
package de.tum.`in`.tumcampusapp.component.tumui.grades.model

import com.tickaroo.tikxml.annotation.Element
import com.tickaroo.tikxml.annotation.Xml

/**
 * Wrapper class holding a list of exams.
 *
 * Note: This model is based on the TUMOnline web service response format for a
 * corresponding request.
 */
@Xml(name = "rowset")
data class ExamList(@Element val exams: List<Exam> = mutableListOf())
